profit_analysis/analysis.py

```diff
@@ -146,174 +146,197 @@ async def get_usd_profit(profit, chain, save_to_csv=False):
                         on=BLOCK_KEY,
                         suffixes=("", "_y"),
                     )
 
-                    category = "liquidation"
-                else:
-                    category = "arbitrage"
-                    profit_with_price_token[PRICE_DEBT_KEY] = 0
-
-                profit_with_price_token[CATEGORY_KEY] = category
-
-                profit_with_price_tokens = pd.concat(
-                    [profit_with_price_tokens, profit_with_price_token]
-                )
-            except Exception as e:
-                print(f"    Failed for token={token}")
-                print(e)
-                failures[token] = e
-    print("Finished processing all tokens")
-    print(f"profit_with_price_tokens=\n{profit_with_price_tokens}")
-    profit_with_price_tokens[PRICE_DEBT_KEY] = profit_with_price_tokens[
-        PRICE_DEBT_KEY
-    ].fillna(value=0)
-    profit_with_price_tokens[AMOUNT_DEBT_KEY] = profit_with_price_tokens[
-        AMOUNT_DEBT_KEY
-    ].fillna(value=0)
-    profit_with_price_tokens[PROFIT_USD_KEY] = (
-        profit_with_price_tokens[AMOUNT_RECEIVED_KEY]
-        * profit_with_price_tokens[PRICE_RECEIVED_KEY]
-        - profit_with_price_tokens[AMOUNT_DEBT_KEY]
-        * profit_with_price_tokens[PRICE_DEBT_KEY]
-    )
-    profit_with_price_tokens = profit_with_price_tokens.reset_index(drop=True)
-    profit_with_price_tokens[DATE_KEY] = profit_with_price_tokens[
-        TIMESTAMP_KEY
-    ].dt.normalize()
+                    if len(debt_tokens_prices) > 0:
+                        debt_tokens_prices[TIMESTAMP_KEY] = pd.to_datetime(
+                            debt_tokens_prices[TIMESTAMP_KEY]
+                        )
+                        profit_with_price_token[BLOCK_KEY] = profit_with_price_token[
+                            BLOCK_KEY
+                        ].astype(int)
+                        debt_tokens_prices[BLOCK_KEY] = debt_tokens_prices[
+                            BLOCK_KEY
+                        ].astype(int)
+                        profit_with_price_token = pd.merge_asof(
+                            profit_with_price_token.sort_values(BLOCK_KEY),
+                            debt_tokens_prices.sort_values(BLOCK_KEY),
+                            direction="nearest",
+                            on=BLOCK_KEY,
+                            suffixes=("", "_y"),
+                        )
+
+                        category = "liquidation"
+                    else:
+                        category = "arbitrage"
+                        profit_with_price_token[PRICE_DEBT_KEY] = 0
+
+                    profit_with_price_token[CATEGORY_KEY] = category
+
+                    profit_with_price_tokens = pd.concat(
+                        [profit_with_price_tokens, profit_with_price_token]
+                    )
+                except Exception as e:
+                    print(f"    Failed for token={token}")
+                    print(e)
```

Comment:
Please combine those two print calls:

```suggestion
                    print(f"    Failed for token={token}, reason={e}")
```